hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,171 @@
+
+
+## Description
+
+This IP block acts as a gasket between requesting peripheral blocks and the CSRNG block.
+One function this IP block performs is to translate data transfer size.
+For example, CSRNG will return 128 bits on the `genbits` bus.
+A peripheral block may connect with a 32-bit data bus.
+The EDN block will move the first 32 bits from the returned `genbits` bus, and hold on to the remaining data until another request ask for more data. 
+Further, if data is not consumed immediately, the interface to the CSRNG will indicate back pressure to the CSRNG block.
+Each of the four interfaces can request data such that any `genbits` bus return can supply any requesting peripheral block.
+
+A configurable feature for peripheral port 0 is that it can be put into broadcast mode.
+This means that the EDN block will request for that port and is paced by a configurable timer.
+Data will be delivered continuously on the bus for any snooping participant to consume.
+
+Commands to the CSRNG block can be generated by either firmware or hardware.
+For firmware generated commands, a TLUL set of registers will allow any CSRNG application interface command to be sent to CSRNG.
+For hardware generated commands, there are two modes that allow this.
+One is the auto request mode, where two FIFOs are provided to send commands and additional data. 
+The general operation of this mode is that the CSRNG instance is set up by firmware, then the FIFOS are preloaded with commands.
+One FIFO can be programmed to send `generate` commands.
+The other FIFO can be programmed to send `reseed` commands.
+There is a configurable register that sets the number of requests between reseeds.
+This register determines how many `generate` commands are sent before a `reseed` command is sent.
+The process of sending these two commands will repeat forever until the module is reset or the enable bit is cleared.
+
+## Example Topology
+The diagram below shows an example topology where two EDN modules are used to distribute CSRNG `genbits` to peripheral modules.
+
+![EDN Example Topology Diagram](edn_top_diag.svg)
+
+
+## Compatibility
+Commands generated toward the CSRNG application interface must conform to the CSRNG application interface specification.
+
+
+# Theory of Operations
+
+The first step that must be done to enable the EDN block is to configure the  {{< regref "CTRL" >}} register.
+This will enable the block in general, and then determine which of three modes are to be enabled.
+Broadcast mode can be selected independent of auto request mode or boot request mode.
+A choice between auto request mode and boot request mode should be made.
+
+Once a CSRNG command request mode has been configured, the EDN block will start requesting to the CSRNG block.
+Data will be returned from the specific CSRNG instance that the EDN block is connected to.
+It is assumed that the EDN block has sent an `instantiate` application command before sending any `generate` application commands.
```

Comment:
Who assumes this? Are you saying that the EDN block hardware ensures this is the case? (In which case, I'm not sure it needs documenting). Or is this about direct CSRNG access through the forwarded register interface?

Reply:
yeah i have the same question :)
Since this could be part of the boot ROM setup.

Reply:
I think I made a comment in the previous PR in this sequence about the interface logic not doing this autonomously but rather the "consumer" connected to the EDN having to ensure this was done - there was some language to that effect.

We could probably clarify this text a bit, if it is truly the obligation of the consumer to first instantiate, then generate.

Reply:
How about this: "The EDN is responsible for instantiating the CSRNG instance before sending generate commands (which the CSRNG requires per NIST SP 800-90A)"

Reply:
Okay, with all the comments on this paragraph I have given it a compete rewrite:

"Once a CSRNG command request mode has been configured, the EDN block will start issuing requests to the CSRNG via a connection to a CSRNG application interface port. The CSRNG application interface implements the "function envelopes" recommended by NIST SP 800-90A for random number generation, and these function envelopes establish certain requirements for the order of operations. For instance, the EDN block sends an `instantiate` command before sending any `generate` commands. Note that once an application interface port has received an `instantiate` command, that application interface port then has access to a unique CSRNG instance, with a unique seed.
Random number consumers on the same EDN share the same CSRNG instance through their common application interface port. Therefore from a security perspective, an attack to that particular CSRNG instance is an attack on the random number stream shared by all consumers on the same EDN. Meanwhile, seeds and other state variables specific to a particular CSRNG instance are not shared between consumers on separate EDN instances, or with any consumers with a connection to a dedicated CSRNG application interface port."

Hopefully that brings out the spirit of this paragraph better.

Reply:
This whole section is rewritten, stay tuned for update.